### PR TITLE
Fix non-visible IRS demo bug

### PR DESCRIPTION
**Pull request by exFalso**
This example code was run as part of the IRS demo

The failure was intermittent because it's run on a codepath that may not notify the initiating flow in time.

### CODE REVIEW

File: samples/irs-demo/cordapp/src/main/kotlin/net/corda/irs/flows/AutoOfferFlow.kt [hunk 1]
```diff
@@ -71,8 +71,8 @@ object AutoOfferFlow {
             // and because in a real life app you'd probably have more complex logic here e.g. describing why the report
             // was filed, checking that the reportee is a regulated entity and not some random node from the wrong
             // country and so on.
-            val regulator = serviceHub.identityService.partiesFromName("Regulator", true).single()
-            subFlow(ReportToRegulatorFlow(regulator, finalTx))
+            // val regulator = serviceHub.identityService.partiesFromName("Regulator", true).single()
```
Comment (josecoll) [inline samples/irs-demo/cordapp/src/main/kotlin/net/corda/irs/flows/AutoOfferFlow.kt]: shouldn't we either add a TODO (to fix at a later date) or remove completely?

Comment (exFalso) [reply samples/irs-demo/cordapp/src/main/kotlin/net/corda/irs/flows/AutoOfferFlow.kt]: from what I understand this is here for documentation purposes (is that correct @mikehearn ?), which is why I left in the commented out code.

Comment (vkolomeyko) [reply samples/irs-demo/cordapp/src/main/kotlin/net/corda/irs/flows/AutoOfferFlow.kt]: Should this PR not be going into `pre-release-3.0-ENT-RC01` branch?

**Comment by vkolomeyko** [comment]
Should this PR not be going into pre-release-3.0-ENT-RC01 branch?

File: samples/irs-demo/cordapp/src/main/kotlin/net/corda/irs/flows/AutoOfferFlow.kt (hunk 1 above)
Comment (mikehearn) [reply samples/irs-demo/cordapp/src/main/kotlin/net/corda/irs/flows/AutoOfferFlow.kt]: It's supposed to actually work. What's the actual fix?

Comment (exFalso) [reply samples/irs-demo/cordapp/src/main/kotlin/net/corda/irs/flows/AutoOfferFlow.kt]: Hmm there is no "Regulator" node... where does that come from?

Comment (mikehearn) [reply samples/irs-demo/cordapp/src/main/kotlin/net/corda/irs/flows/AutoOfferFlow.kt]: That's odd. It should be set up in the gradle file I guess. The node doesn't have to do anything other than have the app loaded. Looking at the commit where I added this I don't see any addition to the gradle file though. I remember testing the app and seeing it work though. I wonder if I forgot to add the gradle file to the commit or something like that.

Can you try just duplicating the node block in the relevant gradle file and see if that fixes it? If not then yes comment it out and I'll take a look at fixing it.

**Comment by exFalso** [comment]
@mikehearn I added the missing node (in Main.kt, build.gradle, IRSDemoTest.kt). Also cherry picked and closed https://github.com/corda/corda/pull/2264

File: samples/irs-demo/cordapp/build.gradle [hunk 2]
```diff
@@ -94,6 +94,14 @@ task deployNodes(type: net.corda.plugins.Cordform, dependsOn: ['jar']) {
         rpcUsers = ext.rpcUsers
         useTestClock true
     }
+    node {
+        name "O=Regulator,L=Moscow,C=RU"
```
Comment (mikehearn) [inline samples/irs-demo/cordapp/build.gradle]: haha, love it

**Pull request merged**